component/src/test/java/io/siddhi/extension/execution/time/UTCTimestampFunctionExtensionTestCase.java

```diff
@@ -83,4 +83,42 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
         AssertJUnit.assertTrue(eventArrived);
         executionPlanRuntime.shutdown();
     }
+
+    @Test
+    public void utcTimestampFunctionExtensionWithCustomDateFormat() throws InterruptedException {
+
+        log.info("utcTimestampFunctionExtensionWithCustomDateFormatTestCase");
+        SiddhiManager siddhiManager = new SiddhiManager();
+
+        String inStreamDefinition = "" +
+                "define stream inputStream (symbol string, price long, format string);";
+        String query = ("@info(name = 'query1') " +
+                "from inputStream " +
+                "select symbol , time:utcTimestamp(format) as utcTimestamp " +
+                "insert into outputStream;");
+        SiddhiAppRuntime executionPlanRuntime = siddhiManager.
+                createSiddhiAppRuntime(inStreamDefinition + query);
+
+        executionPlanRuntime.addCallback("query1", new QueryCallback() {
+            @Override
+            public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
+                EventPrinter.print(timeStamp, inEvents, removeEvents);
+                eventArrived = true;
+                for (Event inEvent : inEvents) {
+                    eventCount.incrementAndGet();
+                    log.info("Event : " + eventCount.get() + ",utcTimestamp : " + inEvent.getData(1));
+                }
+            }
+        });
+
+        InputHandler inputHandler = executionPlanRuntime.getInputHandler("inputStream");
+        executionPlanRuntime.start();
+        inputHandler.send(new Object[]{"IBM", 700f, "yyyy-MM-dd HH:mm:ss.SSS"});
+        inputHandler.send(new Object[]{"WSO2", 60.5f, "yyyy-MM-dd HH:mm:ss"});
+        inputHandler.send(new Object[]{"XYZ", 60.5f, "yyyy-MM-dd HH:mm"});
+        SiddhiTestHelper.waitForEvents(waitTime, 3, eventCount, timeout);
+        AssertJUnit.assertEquals(3, eventCount.get());
+        AssertJUnit.assertTrue(eventArrived);
```

Comment:
Assert that you are getting in the format you requested using a regex

Reply:
fixed in https://github.com/siddhi-io/siddhi-execution-time/pull/62/commits/bbc32f142194c2241f35bd972af3ff0b75661578